rpc/backend/tx_info.go

```diff
@@ -158,6 +159,13 @@ func (b *Backend) GetTransactionReceipt(hash common.Hash) (map[string]interface{
 			break // Found the transaction
 		}
 
+		if attempt == maxRetries/2 && b.Mempool != nil {
```

Comment:
Avoid waiting all retries which cost 100s (around default timeout for web3 client) by early return